pkg/sectigo/server/auth.go

```diff
@@ -0,0 +1,214 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	jwt "github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	"github.com/trisacrypto/directory/pkg/sectigo"
+)
+
+func (s *Server) Login(c *gin.Context) {
+	in := &sectigo.AuthenticationRequest{}
+	if err := c.ShouldBindJSON(in); err != nil {
+		c.JSON(http.StatusBadRequest, Err(err))
+		return
+	}
+
+	// Very basic auth as this is only used for staging
+	if in.Username != s.conf.Auth.Username || in.Password != s.conf.Auth.Password {
+		c.JSON(http.StatusForbidden, Err("could not authenticate user with password"))
+		return
+	}
+
+	var err error
+	out := &sectigo.AuthenticationReply{}
+	if out.AccessToken, out.RefreshToken, err = s.tokens.SignedTokenPair(); err != nil {
+		c.JSON(http.StatusInternalServerError, Err(err))
+		return
+	}
+
+	c.JSON(http.StatusOK, out)
+}
+
+func (s *Server) Refresh(c *gin.Context) {
+	token, err := io.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, Err(err))
+		return
+	}
+
+	if _, err = s.tokens.Verify(string(token)); err != nil {
+		fmt.Println(err)
```

Comment:
I assume this was left in from debugging?

Reply:
good catch; thanks!